azurerm/data_source_api_management.go

```diff
@@ -0,0 +1,336 @@
+package azurerm
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/apimanagement/mgmt/2018-06-01-preview/apimanagement"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceApiManagementService() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceApiManagementRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.ApiManagementServiceName,
+			},
+
+			"resource_group_name": resourceGroupNameForDataSourceSchema(),
+
+			"location": locationForDataSourceSchema(),
+
+			"publisher_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"publisher_email": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"sku": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"capacity": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"notification_sender_email": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"gateway_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"gateway_regional_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"portal_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"management_api_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"scm_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"additional_location": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"location": locationForDataSourceSchema(),
+
+						"gateway_regional_url": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"static_ips": {
```

Comment:
(as below) perhaps this should become `public_ip_addresses` since there can also be Private IP Addresses?